### PR TITLE
chore: add upgrade handler to simapp for v6 -> v7

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #2437 

### Commit Message / Changelog Entry

```bash
chore: add upgrade handler to simapp for v6 -> v7
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: testing/simapp/app.go [hunk 1]
```diff
@@ -891,4 +892,14 @@ func (app *SimApp) setupUpgradeHandlers() {
 			ibcmock.ModuleName+icacontrollertypes.SubModuleName,
 		),
 	)
+
+	app.UpgradeKeeper.SetUpgradeHandler(
```
Comment (chatton) [inline testing/simapp/app.go:896]: [nit] not necessarily related to this PR, but the terminology of `Set` in `SetUpgradeHandler` is kind of confusing, in reality we are adding/appending a new upgrade handler. the name `AddUpgradeHandler` might makes more sense WDYT? We can create a separate issue if people agree with this.

cc @colin-axner @damiannolan

**Review by chatton — APPROVED** [review]
LGTM! Left one comment but not actually related to this PR specifically 👍

File: testing/simapp/app.go (hunk 1 above)
Comment (charleenfei) [reply testing/simapp/app.go:896]: this makes sense to me wrt re-naming!

Comment (colin-axner) [reply testing/simapp/app.go:896]: You'd have to change the [function](https://github.com/cosmos/cosmos-sdk/blob/main/x/upgrade/keeper/keeper.go#L70) on the SDK :smile:

File: testing/simapp/upgrades/v7/upgrades.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		// Perform in-place store migrations for the v7 upgrade
+		// The migration includes:
+		//
+		// - Migrating solo machine client states from v2 to v3 protobuf definition
+		// - Pruning all solo machine consensus states
+		// - Removing the localhost client
+		// - Asserting existing tendermint clients are properly registered on the chain codec
+		if err := v7.MigrateStore(ctx, hostStoreKey, cdc); err != nil {
+			return nil, err
+		}
```
Comment (colin-axner) [inline testing/simapp/upgrades/v7/upgrades.go:36]:
```suggestion
```

These will be performed automatically as a call to this function is called by `Migrate2to3` in the registered migration function for the IBC module bump from "consensus version" 2 to 3

File: testing/simapp/upgrades/v7/upgrades.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+		// OPTIONAL: prune expired tendermint consensus states to save storage space
```
Comment (colin-axner) [inline testing/simapp/upgrades/v7/upgrades.go:38]: :+1:

**Pull request merged**